### ISSUE TITLE
Parsing large TSV (converted to CSV) and writing to XLSX causes some rows of data to be lost

### ISSUE THREAD

**Issue by RSpicer** [SheetJS/js-xlsx#1301]
Hi there! I'll try to give you as much information about my issue as possible.

Basically, we have an export data function that takes raw data retrieved from our database, and exports it to a file in a readable format. The two options we have are TSV (which the function was initially created for) and XLSX. This means that, in the endpoint im calling to get this export, we have a TSV represented as a string. I convert it to CSV and read it in, then export it, with these two lines.

```
var wb = xlsx.read(totalStream.split(',').join('","').split('\t').join(','), { type: "string", raw: true }),
outStream = xlsx.write(wb, { type: "buffer", bookType: 'xlsx' });
```

totalStream in this example is the raw tsv string, with \t seperating columns and values. We're sure the output of this TSV is correct.

The problem is, when handling relatively large files (1000+ rows and 400+ columns), we get less data in the XLSX version when compared to the TSV. For reference, here are the numbers:

TSV: 2128 rows, 430 columns
XLSX: 1198 rows, 430 columns

You should also know that, when opening the XLSX in LibreOffice, I get the following error: 

> Warning loading document {name}.xlsx. The data could not be loaded completely because the maximum number of columns per sheet was exceeded.

Basically, some of our data is missing, and we're not sure why. My understanding is that XLSX has a huge number of max rows, so I doubt it's that. Is there anything that stands out in my process here that seems off, or is this a package bug? Any pointers or steps in the right direction are welcome.

**Comment by SheetJSDev** [comment]
@RSpicer TSV should be autodetected.  The logic involves counting the number of commas and tab characters that are not within a quoted string.  See https://github.com/SheetJS/js-xlsx/blob/master/bits/40_harb.js#L729-L765 for the actual implementation.

Can you just try passing the tabbed string in directly?

As for the LibreOffice error, the message clearly says "maximum number of columns".  It's possible that LibreOffice ingress is limited to 256 columns (which is the XLS limit).  Have you tried opening the file with Excel?  If you don't have access to Excel, please share a file (you can email dev at sheetjs dot com) and we can take a look

**Comment by RSpicer** [comment]
Sorry, just realized that i posted the wrong example. Here's the correct one:

```
var wb = xlsx.read(totalStream.split(',').join('","').split('\t').join(','), { type: "string", raw: true }),
outStream = xlsx.write(wb, { type: "buffer", bookType: 'xlsx' });
```

**Comment by SheetJSDev** [comment]
Can you share a file?  If it is too large for a pastebin-like site, send an email

**Comment by RSpicer** [comment]
This is fixed, and you can close. I used another library for parsing CSVs, as apparently my method of parsing CSVs wasnt working with the TSV output we had. Feeding this CSV output in to XLSX worked like a charm.

My bad, Cheers!

**Comment by SheetJSDev** [comment]
To be clear, if you can't pass the original TSV output to the library directly, then we still have a bug and it would be useful to see the first row where the Excel result diverges from the SheetJS result.

**Comment by RSpicer** [comment]
I'll try to get you an example tomorrow. If you haven't heard from me by
the end of the day, feel free to ping me again.
